File: client/lib/features/onboarding/models/occupation_tag.dart
class OccupationTag {
  final int id;
  final String label;
  final String? icon;
  final String category;

  OccupationTag({
    required this.id,
    required this.label,
    this.icon,
    required this.category,
  });

  factory OccupationTag.fromJson(Map<String, dynamic> json) {
    return OccupationTag(
      id: json['id'],
      label: json['label'],
      icon: json['icon'],
      category: json['category'] ?? 'Occupation',
    );
  }
}

File: server/src/main/java/cn/cuckoox/wisediet/model/OccupationTag.java
package cn.cuckoox.wisediet.model;

import org.springframework.data.annotation.Id;
import org.springframework.data.relational.core.mapping.Table;
import lombok.Data;
import lombok.AllArgsConstructor;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Table("occupation_tags")
public class OccupationTag {
    @Id
    private Long id;
    private String label;
    private String icon;
    private String category;
}
